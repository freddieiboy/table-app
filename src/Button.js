import React, { Component } from 'react';

class Button extends Component {
  constructor(props) {
    super(props);
    this.state = {
      pressed: false
    }
  }

  setAction = () => {
    this.props.action();
    this.unPress();
  }

  press = () => {
    return this.setState({pressed: true});
  }

  unPress = () => {
    return this.setState({pressed: false});
  }

  render() {
    const styles = {
      Button: {
        backgroundColor: 'black',
        borderColor: 'black'
      },
      outline: {
        color: 'black',
        backgroundColor: 'white'
      },
      small: {
        fontSize: '.8rem',
        height: '2.8rem',
        lineHeight: '2.8rem',
        padding: '0 1.5rem'
      },
      pressed: {
        opacity: .6,
        backgroundColor: '#f1f1f1'
      }
    }
    return (
      <button
        className="button button-black"
        style={ifStyle(
          styles.Button,
          this.props.type === 'outline' && styles.outline,
          this.props.size === 'small' && styles.small,
          this.state.pressed && styles.pressed
        )}
        onMouseUp={this.setAction}
        onMouseDown={this.press}>
        {this.props.label}
      </button>
    )
  }
}

export function ifStyle() {
  var res = {};
  for (var i = 0; i < arguments.length; ++i) {
    if (arguments[i]) {
      Object.assign(res, arguments[i]);
    }
  }
  return res;
}

export default Button;
