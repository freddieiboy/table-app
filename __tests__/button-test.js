import React from 'react';
import Button from '../src/Button';
import renderer from 'react-test-renderer';

describe('App container', () => {
  it('should exist', () => {
    const component = renderer.create(
      <Button />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
