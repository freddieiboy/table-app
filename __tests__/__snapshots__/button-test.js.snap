exports[`App container should exist 1`] = `
<button
  className="button button-black"
  onMouseDown={[Function anonymous]}
  onMouseUp={[Function anonymous]}
  style={
    Object {
      "backgroundColor": "black",
      "borderColor": "black"
    }
  } />
`;
